app/src/main/java/com/coveros/coverosmobileapp/blogpost/CommentFormActivity.java

```diff
@@ -65,30 +78,111 @@ public void onClick(View v) {
+            Matcher emailMatcher = emailPattern.matcher(email);
+            boolean isValidEmail = emailMatcher.matches();
+
+            JsonObject body = createCommentRequestBody(postId, author, email, message);
+
+            commentRequest = new RestRequest(COMMENT_URL, null, body, new Response.Listener() {
+                @Override
+                public void onResponse(Object response) {
+                    createSuccessDialog(CommentFormActivity.this);
+                    successDialog.show();
+                }
+            }, new Response.ErrorListener() {
+                @Override
+                public void onErrorResponse(VolleyError error) {
+                    createErrorDialog(CommentFormActivity.this);
+                    errorDialog.show();
+                }
+            });
+
+            boolean hasNoEmptyFields = emptyFields.isEmpty();
+            if (hasNoEmptyFields && isValidEmail) {
+                RequestQueue requestQueue = Volley.newRequestQueue(CommentFormActivity.this);
+                requestQueue.add(commentRequest);
+            } else if (!isFinishing()) {
+                if (!hasNoEmptyFields) {
+                    createEmptyFieldDialog(emptyFields);
+                    emptyFieldDialog.show();
+                } else {  // if not a valid email
+                    createInvalidEmailDialog();
+                    invalidEmailDialog.show();
+                }
+            }
         }
 
-        emptyFieldAlertDialog.setTitle(getResources().getString(R.string.empty_field_alert_dialog_title));
-        emptyFieldAlertDialog.setMessage(getResources().getString(R.string.empty_field_alert_dialog_message) + " " + emptyFieldsString + ".");
-        emptyFieldAlertDialog.setButton(AlertDialog.BUTTON_NEUTRAL, getResources().getString(R.string.empty_field_alert_dialog_dismiss_message),
-                new DialogInterface.OnClickListener() {
-                    @Override
-                    public void onClick(DialogInterface dialog, int which) {
-                        dialog.dismiss();
-                    }
-                });
-        return emptyFieldAlertDialog;
+        private void createSuccessDialog(Context context) {
+            successDialog = new AlertDialog.Builder(context).create();
+            successDialog.setTitle(context.getString(R.string.success_dialog_title));
+            successDialog.setMessage(context.getString(R.string.success_dialog_message));
+            successDialog.setButton(AlertDialog.BUTTON_NEUTRAL, context.getString(R.string.success_dialog_button), new DialogInterface.OnClickListener() {
+                @Override
+                public void onClick(DialogInterface dialog, int which) {
+                    dialog.dismiss();
+                    finish();
+                }
+            });
+        }
+
+        private void createErrorDialog(Context context) {
```

Comment:
I know you are planning on doing some refactoring in another branch, so this might be better suited there, but these AlertDialog creation methods might work better if you actually return the created dialog instead of manipulating an object attribute.